### PR TITLE
Program for minimum Sum of Squares in Java

### PR DESCRIPTION
**Pull request by divyansh-ag-03**
# Language

What language was used for the solution?
I used Java

# Challenge

What is the name of the challenge being completed?
Sum of Squares

Screenshot 2022-10-06 220547
Uploading the Screenshot for the outputs of some test cases.

### CODE REVIEW

**Comment by ericwburden** [comment]
Addresses #348

File: challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            }
+        }
+        return res;
+    }
+
+    public static void main(String[] args) 
+    {   //running tests
+        System.out.println(SumOfSquares(8));
+        System.out.println(SumOfSquares(12));
+        System.out.println(SumOfSquares(9));
```
Comment (ericwburden) [inline challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java:28]: You should pass the test cases presented in the original challenge. They're a _bit_ buried in the test file, but they go like this:

```
SumOfSquares(13)  -> 2
SumOfSquares(27)  -> 3
SumOfSquares(144) -> 1
SumOfSquares(84)   -> 3
SumOfSquares(85)   -> 2
```

Your solution passes the first two, but it hangs on the last three. It _looks_ like it's filling the stack up with recursive calls.

**Review by ericwburden — CHANGES_REQUESTED** [review]

File: challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java (hunk 1 above)
Comment (ericwburden) [reply challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java:28]: I would probably recommend a second function of the form `bool canSum(int n, int[] numbers, int k)` that will take a number `n`, a list of numbers `numbers`, and an addend count `k` and check to see if any `k` numbers in `numbers` can sum to `n`. With that in hand, you should have no trouble checking to see if there's 2, 3, 4, etc. numbers that can sum to `n`.

Comment (divyansh-ag-03) [reply challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java:28]: On it.

Comment (ericwburden) [reply challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java:28]: Personally, I prefer to work on things on my own computer. Unless you deleted your branch, you can just make more commits against it. When you push the commits to your remote, they'll show up on the PR as well.

**Comment by divyansh-ag-03** [comment]
Hi @ericwburden , have changed the code and ran the required test cases. It's running successfully by implementing Dynamic Programming.
Please review and let me know.
Thanks

File: challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        // rest of table using recursion
+        for (int i = 4; i <= n; i++)
+        {
+         
+            // max value is i as i can
+            // always be represented
+            dp[i] = i;
+ 
+            // Going through all smaller numbers to recursively find minimum
```
Comment (ericwburden) [inline challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java:28]: Please update or remove this comment (it's referring to the recursive approach from the first commit on this PR). Don't want to confuse folks who come later to learn from these examples.

**Review by ericwburden — CHANGES_REQUESTED** [review]
Minor change requested on a comment. Otherwise, this code works and is a nice example of DP. Good job!

File: challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java (hunk 2 above)
Comment (divyansh-ag-03) [reply challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java]: Thank you @ericwburden , It's done. Have a look at it.

**Comment by divyansh-ag-03** [comment]
Kindly let me know if any more changes are required. 
Also, merge it to the repo with the 'Hacktober-Accepted' tag if satisfied.
Also thanks for giving me the opportunity to work on it.
Thanks

**Review by drkennetz — APPROVED** [review]
Nice change of direction, DP is a good choice here.

**Comment by drkennetz** [comment]
@divyansh-ag-03 thanks for contributing! We appreciate all solutions and the thought you put into this. You can solve and PR additional challenges and we can add the hacktoberfest-accepted label, even if there is no issue for it on GitHub. If you see any that you'd like to work on, feel free to do so!

**Comment by divyansh-ag-03** [comment]
Thanks for that @drkennetz, will let you know if I found an issue I can work upon.

**Review by ericwburden — APPROVED** [review]
Looks great! Welcome again!

**Pull request merged**